bin/is-prerelease

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# is-prerelease prints "true" if the git tag denotes a prerelease version.
+# Prints "false" otherwise.
+#
+# It handles both simple git tags (starting with "v", e.g v0.2.5 or v4.2.0) and
+# prefixed git tags (e.g "flutter_comms-v0.0.5" or "leancode_lint-v1.3.0").
+#
+# Usage:
+#
+# $ is-prerelease <package_name> <git_tag>
+#
+# For example:
+#
+# $ is-prerelease some_single_package v0.7.3
+#
+# $ is-prerelease leancode_contracts leancode_contracts-v2.1.0
+
+package_name="${1:-}"
+tag="${2:-}"
+
+if [ -z "$package_name" ]; then
+    echo "error: missing package name" 1>&2
+    exit 1
+fi
+
+if [ -z "$tag" ]; then
+    echo "error: missing tag" 1>&2
+    exit 1
+fi
+
+prefix="${package_name}-"
+version="${tag#$prefix}"
+version="${version#v}"
+
+if [[ "$version" == 0* ]]; then
+    echo "true"
+else
+    echo "false"
+fi
```

Comment:
Is using `true` and `false` strings okay?

Reply:
Why not? There is no convention here afaict so the way the tools like this one return their results is all over the place.

* `test` and various similar builtins like `[` and `[[` return `0` as exit code on success and `1` on failure
* [`vercmp`](https://man.archlinux.org/man/vercmp.8) prints `-1` / `0`/ `1` to stdout
* `systemctl is-…` both prints a string and returns a meaningful exit code unless `--quiet` is specified:

```
$ systemctl is-system-running 
degraded

$ echo $?
1
```

My point is, whatever you do, it'll probably be fine.